test/Test/Data/Enum.purs

```diff
@@ -44,6 +45,28 @@ instance boundedEnumT :: BoundedEnum T where
   toEnum = defaultToEnum
   fromEnum = defaultFromEnum
 
+-- | A newtype over Int which is supposed to represent Ints bounded between 0
+-- | and 10,000. Why 10,000? It's just about large enough that we'll most
+-- | likely see stack overflow errors if we've managed to break TCO.
+newtype Upto10k = Upto10k Int
```

Comment:
I've seen stack-unsafe code work up to 30k iterations in the past. I always do 100k.

Reply:
Huh, news to me - good to know, thanks! I'll change this.